src/AutoRest.CSharp/LowLevel/Output/OperationMethodChainBuilder.cs

```diff
@@ -464,6 +472,6 @@ private ReferenceOrConstant CreateReference(InputParameter? operationParameter,
 
         private record ReturnTypeChain(CSharpType Convenience, CSharpType Protocol, CSharpType? ConvenienceResponseType);
 
-        private record ParameterChain(InputParameter? Input, Parameter? Convenience, Parameter? Protocol, Parameter? CreateMessage);
+        private record ParameterChain(Parameter? Convenience, Parameter? Protocol, Parameter? CreateMessage, bool IsSpreadParameter = false);
```

Comment:
ParameterChain contains parameters in different usage senarios, it may be a little confuse to add a `Kind` in such a chain.

Reply:
I did not add a `Kind`. I just spread some property you might need later in this class to decouple the Input out of this

Reply:
You could regard this as a follow up change on this comment: https://github.com/Azure/autorest.csharp/pull/3006#discussion_r1071133460

Reply:
Sorry. I mean it is a little confuse to add a bool `IsSpreadParameter` to a parameter chain.

Reply:
We cannot add it to the `Parameter` because it stands for a parameter of dotnet method in which "spread" or not makes no sense.
We do not want things in `Input` to couple with other things, therefore this is the only option I could think of.
Speaking of confusion, I believe the name of this record is more confusing.

Reply:
Ok. We may refine the name later.